File: src/main/java/com/example/desafio/api/models/entitys/ModelCarEntity.java
package com.example.desafio.api.models.entitys;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Table(name = "model_cars")
@Data
public class ModelCarEntity extends BaseEntity {

    private String modelName;


    @Column(name = "price_fip", precision = 12, scale = 3)
    private BigDecimal priceFip;

    @OneToMany(mappedBy = "modelId", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<CarEntity> cars;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "brand_id")
    private BrandEntity brandId;

    public ModelCarEntity(String modelName, BigDecimal priceFip, List<CarEntity> cars, BrandEntity brandId) {
        this.modelName = modelName;
        this.priceFip = priceFip;
        this.cars = cars;
        this.brandId = brandId;
    }

    public ModelCarEntity() {
    }
}

File: src/main/kotlin/com/example/desafio/api/models/entitys/ModelCarEntity.kt
package com.example.desafio.api.models.entitys

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "model_cars")
data class ModelCarEntity(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        var deleted: Boolean = false,

        @Column(name = "model_name", unique = true)
        var modelName: String? = null,

        @Column(name = "price_fip", precision = 12, scale = 3)
        var priceFip: BigDecimal? = null,

        @OneToMany(mappedBy = "modelId", cascade = [CascadeType.ALL], orphanRemoval = true)
        var cars: List<CarEntity>? = null,

        @JsonIgnore
        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "brand_id")
        var brandId: BrandEntity? = null
)